core/src/test/scala/org/apache/spark/rdd/PairRDDFunctionsSuite.scala

```diff
@@ -524,6 +525,13 @@ class PairRDDFunctionsSuite extends SparkFunSuite with SharedSparkContext {
     pairs.saveAsNewAPIHadoopFile[ConfigTestFormat]("ignored")
   }
 
+  test("The JobId on driver and executor should be the same during the commit") {
+    val pairs = sc.parallelize(Array((new Integer(1), new Integer(2))), 1).
+      map( p => (new Integer(p._1 + 1), new Integer(p._2 + 1))).filter( p => p._1 > 0)
```

Comment:
style nits: 
looks like an extra set of parens inside `Array`?

don't think you need `new Integer`, you can just put the raw ints

if you've got a `=>`  in your call to `map`, `filter`, etc., than use braces, not parens

so

```scala
val pairs = sc.parallelize(Array(1, 2), 1).
    map { p => (p._1 + 1, p._2 + 1) } .filter { p => p._1 > 0 }
```

and add a comment about why you are adding these extra operations before the `save` call.

Reply:
The extra parenthesis is because the array has the type of pairs. I will add another pair to make it clear. I will apply your other changes.

Reply:
The extra parenthesis is because I am trying to build an array of pairs. I will add another pair to make it clear. I will apply your other comments as well. Thank you very much.

Reply:
oh, of course on the pairs, sorry I missed that.